javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseScenario.java

```diff
@@ -0,0 +1,438 @@
+            making it easier to analyze and gain insights from the data. AWS IoT SiteWise also provides tools for 
+            visualizing the data, setting up alarms and alerts, and generating reports.
+                                
+            Another key feature of AWS IoT SiteWise is its ability to scale to handle large volumes of data. 
+            It can collect and store data from thousands of devices and process millions of data points per second, 
+            making it suitable for large-scale industrial operations. Additionally, AWS IoT SiteWise is designed 
+            to be secure and compliant, with features like role-based access controls, data encryption, 
+            and integration with other AWS services for additional security and compliance features.
+                        
+            Let's get started...
+            """);
+
+        waitForInputToContinue(scanner);
+        logger.info(DASHES);
+
+        try {
+            runScenario(assetModelName, assetName, portalName, contactEmail, gatewayName, myThing);
+        } catch (RuntimeException e) {
+           logger.info(e.getMessage());
+        }
+    }
+
+    public static void runScenario(String assetModelName, String assetName,  String portalName, String contactEmail, String gatewayName, String myThing) throws Throwable {
+        logger.info("Use AWS CloudFormation to create an IAM role that is required for this scenario.");
+        CloudFormationHelper.deployCloudFormationStack(ROLES_STACK);
+        Map<String, String> stackOutputs = CloudFormationHelper.getStackOutputs(ROLES_STACK);
+        String iamRole = stackOutputs.get("SitewiseRoleArn");
+        logger.info("The ARN of the IAM role is {}",iamRole);
+        logger.info(DASHES);
+
+        logger.info(DASHES);
+        logger.info("1. Create an AWS SiteWise Asset Model");
+        logger.info("""
+             An AWS IoT SiteWise Asset Model is a way to represent the physical assets, such as equipment,
+             processes, and systems, that exist in an industrial environment. This model provides a structured and
+             hierarchical representation of these assets, allowing users to define the relationships and properties
+             of each asset.
+             
+             This scenario creates two asset model properties: temperature and humidity.
+            """);
+        waitForInputToContinue(scanner);
+        String assetModelId;
+        try {
+            CompletableFuture<CreateAssetModelResponse> future = sitewiseActions.createAssetModelAsync(assetModelName);
+            CreateAssetModelResponse response = future.join();
+            assetModelId = response.assetModelId();
+            logger.info("Asset Model successfully created. Asset Model ID: {}. ", assetModelId);
+
+        } catch (RuntimeException rt) {
+            Throwable cause = rt;
+            while (cause.getCause() != null && !(cause instanceof ResourceAlreadyExistsException)) {
```

Comment:
`while` statement? join will only ever throw a CompletionException with the actual as the cause, unless we wrap the cause in another exception. I guess we need to be consistent about this.

Reply:
Lets discuss